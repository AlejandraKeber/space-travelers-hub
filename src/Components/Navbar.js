import React from 'react';
import { NavLink } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { getMissions } from '../redux/missions/missions';
import icon from '../img/icon.png';
import '../css/Navbar.css';

export default function Navbar() {
  const dispatch = useDispatch();
  return (
    <nav className="flex">
      <div className="logo flex">
        <img src={icon} alt="Logo" />
        <h1>Space Travelers&apos; Hub</h1>
      </div>
      <ul className="flex">
        <li>
          <NavLink
            className="link"
            to="/"
            style={({ isActive }) => (isActive ? { textDecoration: 'underline' } : {})}
          >
            Rockets
          </NavLink>
        </li>
        <li>
          <NavLink
            onClick={() => { dispatch(getMissions()); }}
            className="link"
            to="/missions"
            style={({ isActive }) => (isActive ? { textDecoration: 'underline' } : {})}
          >
            Missions
          </NavLink>
        </li>
        <li>
          <NavLink
            className="link"
            to="/myprofile"
            style={({ isActive }) => (isActive ? { textDecoration: 'underline' } : {})}
          >
            My Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  );
}
