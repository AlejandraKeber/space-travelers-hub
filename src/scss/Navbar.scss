* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.flex {
  display: flex;
}

nav {
  justify-content: space-between;
  align-items: center;
  padding: 1rem;

  img {
    width: 80px;
  }

  .logo {
    align-items: center;
    gap: 1rem;
  }

  ul {
    list-style: none;
    gap: 1rem;
  }

  .link {
    text-decoration: none;
    color: blue;
  }
}
