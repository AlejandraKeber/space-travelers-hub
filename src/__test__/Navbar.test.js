import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Navbar from '../Components/Navbar';

describe('Navbar component', () => {
  test('Test Navbar render correctly', () => {
    const navbar = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    );
    expect(navbar).toMatchSnapshot();
  });
});
