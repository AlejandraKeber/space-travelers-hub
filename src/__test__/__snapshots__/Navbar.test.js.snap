// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Test Navbar render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="flex"
      >
        <div
          class="logo flex"
        >
          <img
            alt="Logo"
            src="icon.png"
          />
          <h1>
            Space Travelers' Hub
          </h1>
        </div>
        <ul
          class="flex"
        >
          <li>
            <a
              aria-current="page"
              class="link active"
              href="/"
              style="text-decoration: underline;"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/myprofile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="flex"
    >
      <div
        class="logo flex"
      >
        <img
          alt="Logo"
          src="icon.png"
        />
        <h1>
          Space Travelers' Hub
        </h1>
      </div>
      <ul
        class="flex"
      >
        <li>
          <a
            aria-current="page"
            class="link active"
            href="/"
            style="text-decoration: underline;"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/myprofile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
